app/domain/authentication/authn_jwt/signing_key/create_signing_key_interface.rb

```diff
@@ -0,0 +1,75 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # Factory that returns the interface implementation of FetchSigningKey
+      CreateSigningKeyInterface ||= CommandClass.new(
+        dependencies: {
+          fetch_provider_uri: Authentication::AuthnJwt::SigningKey::FetchProviderUriSigningKey,
+          fetch_jwks_uri: Authentication::AuthnJwt::SigningKey::FetchJwksUriSigningKey,
+          logger: Rails.logger
+        },
+        inputs: %i[authentication_parameters]
+      ) do
+        def call
+          create
+        end
+
+        private
+
+        def create
+          @logger.debug(LogMessages::Authentication::AuthnJwt::SelectingSigningKeyInterface.new)
+          validate_key_configuration
+
+          if provider_uri_has_valid_configuration?
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedSigningKeyInterface.new(PROVIDER_URI_INTERFACE_NAME))
+            fetch_provider_uri_signing_key
+          elsif jwks_uri_has_valid_configuration?
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedSigningKeyInterface.new(JWKS_URI_INTERFACE_NAME))
+            fetch_jwks_uri_signing_key
+          end
+        end
+
+        def validate_key_configuration
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatingJwtSigningKeyConfiguration.new)
+          if (provider_uri_has_valid_configuration? && jwks_uri_has_valid_configuration?) ||
+              (!provider_uri_has_valid_configuration? && !jwks_uri_has_valid_configuration?)
+            raise Errors::Authentication::AuthnJwt::InvalidUriConfiguration.new(
+              PROVIDER_URI_RESOURCE_NAME,
+              JWKS_URI_RESOURCE_NAME
+            )
+          end
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedSigningKeyConfiguration.new)
+        end
+
+        def provider_uri_has_valid_configuration?
+          return @provider_uri_has_valid_configuration unless @provider_uri_has_valid_configuration.nil?
+          @provider_uri_has_valid_configuration ||= fetch_provider_uri_signing_key.valid_configuration?
+        end
+
+        def jwks_uri_has_valid_configuration?
+          return @jwks_uri_has_valid_configuration unless @jwks_uri_has_valid_configuration.nil?
+          @jwks_uri_has_valid_configuration ||= fetch_jwks_uri_signing_key.valid_configuration?
+        end
```

Comment:
please see this: https://github.com/cyberark/conjur/pull/2227/files?file-filters%5B%5D=.feature&file-filters%5B%5D=.rb&file-filters%5B%5D=.yml&file-filters%5B%5D=No+extension#r648215472

Reply:
DOne with defined?